### PR TITLE
Set xattr before rename to make blob create atomic

### PR DESCRIPTION
**Pull request by andrewgaul**
Previously concurrent operations could expose an object before put had
completed.  Note that the temporary file is still exposed to the
client.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@kahing Please review.

File: apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java [hunk 1]
```diff
@@ -475,20 +476,22 @@ public String putBlob(final String containerName, final Blob blob) throws IOExce
...
-               logger.debug("xattrs not supported on %s", outputPath);
+               logger.debug("xattrs not supported on %s", tmpPath);
             }
          }
-         setBlobAccess(containerName, blobKey, BlobAccess.PRIVATE);
+
+         setBlobAccess(containerName, tmpBlobName, BlobAccess.PRIVATE);
+
+         if (!tmpFile.renameTo(outputFile)) {
+            throw new RuntimeException("Could not rename file " + tmpFile + " to " + outputFile);
```
Comment (kahing) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java]: if rename fails, tmpFile should be cleaned up

**Comment by kahing** [comment]
:+1:

File: apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java (hunk 1 above)
Comment (andrewgaul) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java]: Existing code cleans up `tmpFile` on line 499.

**Pull request merged**